src/main/java/com/byznass/tiolktrack/postgres/dao/PostgresGpsProvider.java

```diff
@@ -31,42 +32,44 @@ public PostgresGpsProvider(Connection connection, LocationProvider locationProvi
 	}
 
 	@Override
-	public Gps getGpsById(String gpsId) throws NoGpsWithIdException, TiolkTrackException {
+	public GpsWithLocations getGpsWithLocations(String userID, String name) throws NoSuchGpsException, TiolkTrackException {
 
-		LOGGER.info("Retrieving Gps entity with id=\'{}\' from postgres database", gpsId);
-		checkIfExists(gpsId);
-		List<Location> locations = locationProvider.getLocationsForGps(gpsId);
-		LOGGER.info("Successfully retrieved Gps entity with id=\'{}\' from postgres database", gpsId);
+		LOGGER.info("Retrieving GpsWithLocations entity (\'{},{}\') from postgres database", userID, name);
+		Gps gps = getGps(userID, name);
+		List<Location> locations = locationProvider.getLocationsForGps(userID, name);
+		LOGGER.info("Successfully retrieved GpsWithLocations entity (\'{},{}\') from postgres database", userID, name);
 
-		return new Gps(gpsId, locations);
+		return new GpsWithLocations(gps, locations);
 	}
 
 	@Override
-	public boolean exists(String gpsId) throws TiolkTrackException {
+	public Gps getGps(String userID, String name) throws NoSuchGpsException, TiolkTrackException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.byznass.tiolktrack.kernel.dao.NoSuchGpsException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.tiolktrack.byznass.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.byznass.tiolktrack.kernel.TiolkTrackException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.tiolktrack.byznass.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)